web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -22,10 +24,99 @@ import {GroundPinService} from 'app/services/ground-pin/ground-pin.service';
 import {DomSanitizer, SafeUrl} from '@angular/platform-browser';
 import {MatDialog} from '@angular/material/dialog';
 import {DataStoreService} from 'app/services/data-store/data-store.service';
+import {LocationOfInterestService} from 'app/services/loi/loi.service';
 import {NavigationService} from 'app/services/navigation/navigation.service';
 import {environment} from 'environments/environment';
-import {Subscription} from 'rxjs';
+import {BehaviorSubject, shareReplay, Subscription, merge, Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
 import {SurveyService} from 'app/services/survey/survey.service';
+import {SubmissionService} from 'app/services/submission/submission.service';
+import {Map as ImmutableMap} from 'immutable';
+
+/** Flat node with expandable and level information */
+export class DynamicFlatNode {
```

Comment:
Since this is a non-trivial class can we move it out to its own file in the same directory?